osu.Game/Screens/Edit/ExternalEditScreen.cs

```diff
@@ -0,0 +1,222 @@
+        {
+            base.LoadComplete();
+
+            timeLoaded = Time.Current;
+
+            fileMountOperation.ContinueWith(t =>
+            {
+                EditOperation = t.GetResultSafely();
+
+                Scheduler.AddDelayed(() =>
+                {
+                    flow.Children = new Drawable[]
+                    {
+                        new OsuSpriteText
+                        {
+                            Text = "Beatmap is mounted externally",
+                            Font = OsuFont.Default.With(size: 30),
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre,
+                        },
+                        new OsuTextFlowContainer
+                        {
+                            Padding = new MarginPadding(5),
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre,
+                            Width = 350,
+                            AutoSizeAxes = Axes.Y,
+                            Text = "Any changes made to the exported folder will be imported to the game, including file additions, modifications and deletions.",
+                        },
+                        new PurpleRoundedButton
+                        {
+                            Text = "Open folder",
+                            Width = 350,
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre,
+                            Action = open,
+                            Enabled = { Value = false }
+                        },
+                        new DangerousRoundedButton
+                        {
+                            Text = "Finish editing and import changes",
+                            Width = 350,
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.TopCentre,
+                            Action = finish,
+                            Enabled = { Value = false }
+                        }
+                    };
+
+                    Scheduler.AddDelayed(() =>
+                    {
+                        foreach (var b in flow.ChildrenOfType<RoundedButton>())
+                            b.Enabled.Value = true;
+                        open();
+                    }, 1000);
+                }, Math.Max(0, 1000 - (Time.Current - timeLoaded)));
+            });
+        }
+
+        private void open()
+        {
+            if (EditOperation == null)
+                return;
+
+            // Ensure the trailing separator is present in order to show the folder contents.
+            gameHost.OpenFileExternally(EditOperation.MountedPath.TrimDirectorySeparator() + Path.DirectorySeparatorChar);
+        }
+
+        public override bool OnExiting(ScreenExitEvent e)
+        {
+            if (!fileMountOperation.IsCompleted)
+                return false;
```

Comment:
If this is hit because the user exits out before the mount completes, there will not be any attempt of cleaning up the temp folder, whatever state it may be in. Not sure how to feel about that.

Reply:
`return false` will block exit, which was intended to handle this case.

Reply:
okay i think i got the return values back to front 😅